manubot/cite/util.py

```diff
@@ -244,18 +249,69 @@ def citation_to_citeproc(citation, prune=True):
     """
     Return a dictionary with citation metadata
     """
-    assert citation == standardize_citation(citation)
+    citation == standardize_citation(citation, warn_if_changed=True)
     source, identifier = citation.split(':', 1)
 
     if source in citeproc_retrievers:
         citeproc_retriever = import_function(citeproc_retrievers[source])
-        citeproc = citeproc_retriever(identifier)
+        csl_item = citeproc_retriever(identifier)
     else:
         msg = f'Unsupported citation source {source} in {citation}'
         raise ValueError(msg)
 
+    from manubot import __version__ as manubot_version
+    from manubot.cite.citeproc import (
+        citeproc_passthrough,
+        append_to_csl_item_note,
+    )
+
+    note_text = f'This CSL JSON Item was automatically generated by Manubot v{manubot_version} using citation-by-identifier.'
+    note_dict = {
+        'standard_citation': citation,
+    }
+    append_to_csl_item_note(csl_item, note_text, note_dict)
+
     citation_id = get_citation_id(citation)
-    from manubot.cite.citeproc import citeproc_passthrough
-    citeproc = citeproc_passthrough(citeproc, set_id=citation_id, prune=prune)
+    csl_item = citeproc_passthrough(csl_item, set_id=citation_id, prune=prune)
+
+    return csl_item
 
-    return citeproc
+
+def csl_item_set_standard_citation(csl_item):
+    """
+    Modify csl_item to set standard_citation. Uses explicit standard_citation if available, either as
+    a standard_citation field or as a key-value pair set using cheater syntax in the note field.
+    Otherwise, a standard_citation is inferred from the id. In both cases, the standard_citation
+    is checked for actually being standard.
+    """
+    from manubot.cite.citeproc import (
+        append_to_csl_item_note,
+        parse_csl_item_note,
+    )
+    if not isinstance(csl_item, dict):
+        raise ValueError("csl_item must be a CSL Data Item represented as a Python dictionary")
+    if 'standard_citation' in csl_item:
+        csl_item['standard_citation'] = standardize_citation(csl_item['standard_citation'], warn_if_changed=True)
+        return csl_item
+    note_dict = parse_csl_item_note(csl_item.get('note', ''))
+    if 'standard_citation' in note_dict:
+        csl_item['standard_citation'] = standardize_citation(note_dict['standard_citation'], warn_if_changed=True)
+        return csl_item
+    if 'id' not in csl_item:
+        raise ValueError('standard_citation cannot be inferred unless the CSL Item id field is set')
+    csl_id = csl_item['id']
+    prefixes = [f'{x}:' for x in list(citeproc_retrievers) + ['raw']]
+    for prefix in prefixes:
+        if csl_id.startswith(prefix):
+            standard_citation = csl_id
+            break
+    else:
+        standard_citation = f'raw:{csl_id}'
+    is_valid_citation_string('@' + standard_citation, allow_raw=True)
```

Comment:
I change this to an assert statement in b3d89c7f37b29d39c901e630ffe76bf38fea5a01. By time `original_citation` reaches this spot (with latest code), a `raw:` prefix should have been added if needed. Therefore, this can only happen when `standard_citation` is explicitly provided and is invalid, in which case, having this function throw an error makes sense.